simpleLooping/normalizeString.py

```diff
@@ -18,3 +18,26 @@
 #   normalizeString("")                  # return ""
 #   normalizeString("Hello  World!")     # return "Hello World!"
 #   normalizeString("  Hello  World!  ") # return "Hello World!"
+from typing import List
+
+def normalizeString(string: str) -> str:
+    lenLst = len(string)
+
+    binary = 0
+    while binary < lenLst and string[binary] == ' ':
+        binary += 1
+
+    normalized = ''
+    if binary < lenLst:
+        normalized += string[binary]
+        binary += 1
+
+    while binary < lenLst:
+        if string[binary] != ' ' or string[binary - 1] != ' ':
+            normalized += string[binary]
+        binary += 1
+
+    if normalized[:-1] == ' ':
+        normalized = normalized[:-1]
+
+    return normalized
```

Comment:
Now think about it, here is my approach if you care. Actually this approach is inspire by your code.

lenStr = len(string)
if lenStr == 0:
    return ""

normalizedString = string[0]
i = 1
while i < lenStr:
    if string[binary] != ' ' or string[binary - 1] != ' ':
        normalized += string[binary]
        i += 1

hasSpaceAtFront = 1 if string[0] == ' ' else 0
hasSpaceAtEnd = 1 if string[-1] == ' ' else 0
normalizedString = normalizedString[hasSpaceAtFront, hasSpaceAtEnd]

return normalizedString